sdk/spring/spring-cloud-azure-feature-management/src/main/java/com/azure/spring/cloud/feature/management/implementation/timewindow/recurrence/RecurrenceEvaluator.java

```diff
@@ -0,0 +1,848 @@
+                / Duration.ofDays((long) interval * RecurrenceConstants.WEEK_DAY_NUMBER).toSeconds();
+            previousOccurrence = alignedStart.plusDays(totalDaysOfFirstInterval
+                + numberOfInterval * interval * RecurrenceConstants.WEEK_DAY_NUMBER);
+            numberOfOccurrences += numberOfInterval * pattern.getDaysOfWeek().size();
+
+            // count the number of occurrence in the first week
+            ZonedDateTime dateTime = alignedStart;
+            while (getDayOfWeek(dateTime) != firstDayOfWeek) {
+                final ZonedDateTime tempDateTime = dateTime;
+                if (pattern.getDaysOfWeek().stream().anyMatch(day ->
+                    convertToWeekDayNumber(day) == getDayOfWeek(tempDateTime))) {
+                    numberOfOccurrences += 1;
+                }
+                dateTime = dateTime.plusDays(1);
+            }
+        } else { // time is still within the first interval
+            previousOccurrence = alignedStart;
+        }
+
+        ZonedDateTime loopedDateTime = previousOccurrence;
+        final ZonedDateTime alignedTime = now.withZoneSameInstant(zoneId);
+        while (!loopedDateTime.isAfter(alignedTime)) {
+            if (!previousOccurrence.isEqual(loopedDateTime) && getDayOfWeek(loopedDateTime) == firstDayOfWeek) {   // Come to the next week
+                break;
+            }
+
+            final ZonedDateTime tempDateTime = loopedDateTime;
+            if (pattern.getDaysOfWeek().stream().anyMatch(day ->
+                convertToWeekDayNumber(day) == getDayOfWeek(tempDateTime))) {
+                previousOccurrence = loopedDateTime;
+                numberOfOccurrences += 1;
+            }
+            loopedDateTime = loopedDateTime.plusDays(1);
+        }
+        return new OccurrenceInfo(previousOccurrence, numberOfOccurrences);
+    }
+
+    /**
+     * Find the closest previous recurrence occurrence before the provided time stamp according to the "AbsoluteMonthly" recurrence pattern.
+     *
+     * @return The return result contains two property, one is previousOccurrence, the other is numberOfOccurrences.
+     * previousOccurrence: The closest previous occurrence.
+     * numberOfOccurrences: The number of complete recurrence intervals which have occurred between the time and the recurrence start.
+     */
+    private OccurrenceInfo getAbsoluteMonthlyPreviousOccurrence() {
+        final RecurrencePattern pattern = settings.getRecurrence().getPattern();
+        final ZonedDateTime start = settings.getStart();
+        final int interval = pattern.getInterval();
+
+        final ZoneId zoneId = getRecurrenceTimeZoneId();
+        final ZonedDateTime alignedStart = start.withZoneSameInstant(zoneId);
+        final ZonedDateTime alignedTime = now.withZoneSameInstant(zoneId);
+
+        int monthGap = (alignedTime.getYear() - alignedStart.getYear()) * 12 + alignedTime.getMonthValue() - alignedStart.getMonthValue();
```

Comment:
I'm assuming this 12 should be a constant named like NUMBER_OF_MONTHS_IN_YEAR, or something?

Reply:
Yes. Add a `NUMBER_OF_MONTHS_IN_YEAR` const to avoid magic number.